doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -81,78 +65,75 @@ Flags, Options and Tables
 Coq has many settings to control its behavior.  Setting types include flags, options
 and tables:
 
-* A :production:`flag` has a boolean value, such as :flag:`Asymmetric Patterns`.
-* An :production:`option` generally has a numeric or string value, such as :opt:`Firstorder Depth`.
-* A :production:`table` contains a set of strings or qualids.
+* A *flag* has a boolean value, such as :flag:`Asymmetric Patterns`.
+* An *option* generally has a numeric or string value, such as :opt:`Firstorder Depth`.
+* A *table* contains a set of strings or qualids.
 * In addition, some commands provide settings, such as :cmd:`Extraction Language`.
 
 .. FIXME Convert "Extraction Language" to an option.
 
 Flags, options and tables are identified by a series of identifiers, each with an initial
 capital letter.
 
-.. cmd::  Set @flag
+.. cmd:: Set @setting_name {? {| @int | @string } }
    :name: Set
 
-   Sets :token:`flag` on.
+   .. insertprodn setting_name setting_name
 
-.. cmd:: Unset @flag
-   :name: Unset
-
-   Sets :token:`flag` off.
+   .. prodn::
+      setting_name ::= {+ @ident }
 
-.. cmd:: Test @flag
+   If :n:`@setting_name` is a flag, no value may be provided; the flag
+   is set to on.
+   If :n:`@setting_name` is an option, a value of the appropriate type
+   must be provided; the option is set to the specified value.
 
-   Prints the current value of :token:`flag`.
+   This command supports the :attr:`local`, :attr:`global` and :attr:`export` attributes.
+   They are described :ref:`here <set_unset_scope_qualifiers>`.
 
+   .. exn:: There is no option @setting_name.
```

Comment:
This is a warning:
```suggestion
   .. warn:: There is no option @setting_name.
```

Reply:
What's the practical difference for the user between a warning and an error?  We know what it means when compiling OCaml--error means the operation (compile) fails, but warnings don't necessarily cause a failure.  By that logic, shouldn't this be an error since the operation fails?

Reply:
> What's the practical difference for the user between a warning and an error?

This means that the file containing such commands is still valid. This is not an error on purpose because sometimes you need to use some compatibility options to make a file compile with several versions of Coq, but this compatibility option might not exist in all these versions. Cf. https://github.com/coq/coq/blob/c5c8ce135606f311834d4c0b9ac3e72be5ee4a36/library/goptions.ml#L356-L358